src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/CollectionControllerTest.php

```diff
@@ -1210,6 +1228,155 @@ public function testDeleteById()
+                    'resourceKey' => MediaInterface::RESOURCE_KEY,
+                ],
+            ],
+        ], $response->children);
+    }
+
+    public function testDeleteByIdWithChildrenWithoutPermissions(): void
+    {
+        $role = $this->createRole('User', 'Sulu');
+
+        $userRole = new UserRole();
+        $userRole->setUser($this->getTestUser());
+        $userRole->setLocale('["en-gb", "de"]');
+        $userRole->setRole($role);
+        $this->em->persist($userRole);
+
+        $this->getTestUser()->addUserRole($userRole);
+        $this->em->flush();
+
+        $permissions = [
+            $role->getId() => [
+                'view' => true,
+                'edit' => true,
+                'add' => true,
+                'delete' => false,
+                'archive' => true,
+                'live' => true,
+                'security' => true,
+            ],
+        ];
+
+        $fullPermissions = [
+            $role->getId() => [
+                'view' => true,
+                'edit' => true,
+                'add' => true,
+                'delete' => true,
+                'archive' => true,
+                'live' => true,
+                'security' => true,
+            ],
+        ];
+
+        $collection = $this->createCollection($this->collectionType1);
+        $collectionId = $collection->getId();
+
+        $child1 = $this->createCollection($this->collectionType1, ['de' => 'Child 1'], $collection);
+        $this->accessControlManager->setPermissions(Collection::class, (string) $child1->getId(), $permissions);
+
+        $child11 = $this->createCollection($this->collectionType1, ['de' => 'Child 1-1'], $child1);
+        $this->accessControlManager->setPermissions(Collection::class, (string) $child11->getId(), $fullPermissions);
+
+        $child111 = $this->createCollection($this->collectionType1, ['de' => 'Child 1-1-1'], $child11);
+        $this->accessControlManager->setPermissions(Collection::class, (string) $child111->getId(), $permissions);
+
+        $child12 = $this->createCollection($this->collectionType1, ['de' => 'Child 1-2'], $child1);
+        $this->accessControlManager->setPermissions(Collection::class, (string) $child12->getId(), $permissions);
+
+        $child2 = $this->createCollection($this->collectionType1, ['de' => 'Child 2'], $collection);
+        $this->accessControlManager->setPermissions(Collection::class, (string) $child2->getId(), $permissions);
+
+        $this->em->clear();
+
+        $this->client->jsonRequest('DELETE', '/api/collections/' . $collectionId);
+        $this->assertHttpStatusCode(403, $this->client->getResponse());
+
+        $response = \json_decode((string) $this->client->getResponse()->getContent());
+        $this->assertEquals(DeletionImpossibleChildPermissionsException::EXCEPTION_CODE, $response->code);
+        $this->assertTrue(isset($response->message));
+        $this->assertObjectHasAttribute('totalUnauthorizedChildren', $response);
```

Comment:
same as above would jsut assertSame the pretty printed json and maybe remove before the stacktrace from it.